### ISSUE TITLE
Help: WARN unbound dns over tls setup: cannot GET content of URL

### ISSUE THREAD

**Issue by AlexAlbright** [qdm12/private-internet-access-docker#156]
Upon `docker-compose up ` the container begins to create successfully then runs into errors when configuring dns.

I've been able to get rid of this error when setting `DOT_IPV6=off`, so it seems there is a DNS over TLS issue specific to IPV6. 

Anyone run into this before?

1. Is this urgent?

    - [ ] Yes
    - [x] No

2. What VPN service provider are you using?

    - [x] PIA
    - [ ] Mullvad
    - [ ] Windscribe

3. What's the version of the program?

    `Running version latest built on 2020-03-13T01:30:06Z (commit d0f678c)`

4. What are you using to run the container?

    - [ ] Docker run
    - [x] Docker Compose
    - [ ] Kubernetes
    - [ ] Docker stack
    - [ ] Docker swarm
    - [ ] Podman
    - [ ] Other:

5. Extra information

Logs:

```pia    | 2020-05-08T15:55:52.982Z       INFO    openvpn: Fri May  8 15:55:52 2020 Initialization Sequence Completed
pia    | 2020-05-08T15:55:52.982Z       INFO    dns configurator: using DNS address 2606:4700:4700::1111 internally
pia    | 2020-05-08T15:55:52.982Z       INFO    dns configurator: using DNS address 2606:4700:4700::1111 system wide
pia    | 2020-05-08T15:55:52.987Z       INFO    Tunnel IP is 104.200.151.89, see more information at https://ipinfo.io/104.200.151.89
pia    | 2020-05-08T15:55:52.988Z       INFO    dns configurator: downloading root hints from https://raw.githubusercontent.com/qdm12/files/master/named.root.updated
pia    | 2020-05-08T15:55:52.995Z       WARN    unbound dns over tls setup: cannot GET content of URL https://raw.githubusercontent.com/qdm12/files/master/named.root.updated: Get "https://raw.githubusercontent.com/qdm12/files/master/named.root.updated": dial tcp: lookup raw.githubusercontent.com on [2606:4700:4700::1111]:53: dial udp [2606:4700:4700::1111]:53: connect: cannot assign requested address
pia    | 2020-05-08T15:55:52.995Z       INFO    dns configurator: downloading root hints from https://raw.githubusercontent.com/qdm12/files/master/named.root.updated
pia    | 2020-05-08T15:55:52.996Z       WARN    unbound dns over tls setup: cannot GET content of URL https://raw.githubusercontent.com/qdm12/files/master/named.root.updated: Get "https://raw.githubusercontent.com/qdm12/files/master/named.root.updated": dial tcp: lookup raw.githubusercontent.com on [2606:4700:4700::1111]:53: dial udp [2606:4700:4700::1111]:53: connect: cannot assign requested address
pia    | 2020-05-08T15:55:52.996Z       INFO    dns configurator: downloading root hints from https://raw.githubusercontent.com/qdm12/files/master/named.root.updated
pia    | 2020-05-08T15:55:52.997Z       WARN    unbound dns over tls setup: cannot GET content of URL https://raw.githubusercontent.com/qdm12/files/master/named.root.updated: Get "https://raw.githubusercontent.com/qdm12/files/master/named.root.updated": dial tcp: lookup raw.githubusercontent.com on [2606:4700:4700::1111]:53: dial udp [2606:4700:4700::1111]:53: connect: cannot assign requested address

```

Configuration file:

```version: "3.7"
services:
  pia:
    image: qmcgaw/private-internet-access
    container_name: pia
    cap_add:
      - NET_ADMIN
    init: true
    ports:
      - 8112:8112/tcp #Torrent Ports
      - 53160:53160/udp
      - 53160:53160/tcp
      - 58846:58846/tcp
    # command:
    environment:
      # More variables are available, see the readme table
      - VPNSP=private internet access
      - USER=<my username>
      - PROTOCOL=udp
      - OPENVPN_VERBOSITY=1
      - OPENVPN_ROOT=no

      # PIA and Windscribe only
      - REGION=US West

      # PIA only
      - PASSWORD=<my password>
      - PIA_ENCRYPTION=strong
      - PORT_FORWARDING=off

      # DNS over TLS
      - DOT=on
      - DOT_PROVIDERS=cloudflare
      - DOT_IPV6=on
      - DOT_VERBOSITY=1
      - BLOCK_MALICIOUS=on
      - BLOCK_SURVEILLANCE=off
      - BLOCK_ADS=off
      - UNBLOCK=off
    restart: always

```

Host OS:

Linux 4.14.24-qnap #1 SMP Fri Apr 10 08:52:15 CST 2020 x86_64 GNU/Linux

**Comment by qdm12** [comment]
Thanks for the detailed debugging. Do you have ipv6 enabled on your host+network+router? If not, it behaves as expected as the ipv6 address of the DNS provider cannot be reached. I'll leave this open to add a check to avoid this error loop. Thanks!

**Comment by AlexAlbright** [comment]
I've disabled IPv6 on my router so that is the issue, thanks for your help!

I am running into a separate issue though, where the the openvpn service seems to restart constantly.

`2020-05-08T18:07:36.608Z        ERROR   openvpn: signal: killed
2020-05-08T18:07:36.608Z        INFO    openvpn: starting
2020-05-08T18:07:36.608Z        INFO    openvpn configurator: starting openvpn
2020-05-08T18:07:36.609Z        ERROR   close |0: file already closed
2020-05-08T18:07:36.615Z        INFO    openvpn: Fri May  8 18:07:36 2020 OpenVPN 2.4.8 x86_64-alpine-linux-musl [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [MH/PKTINFO] [AEAD] built on Feb  7 2020
2020-05-08T18:07:36.616Z        INFO    openvpn: Fri May  8 18:07:36 2020 library versions: OpenSSL 1.1.1g  21 Apr 2020, LZO 2.10
2020-05-08T18:07:36.624Z        INFO    openvpn: Fri May  8 18:07:36 2020 TCP/UDP: Preserving recently used remote address: [AF_INET]104.200.151.74:1197
2020-05-08T18:07:36.624Z        INFO    openvpn: Fri May  8 18:07:36 2020 UDP link local: (not bound)
2020-05-08T18:07:36.624Z        INFO    openvpn: Fri May  8 18:07:36 2020 UDP link remote: [AF_INET]104.200.151.74:1197
2020-05-08T18:07:36.624Z        INFO    openvpn: Fri May  8 18:07:36 2020 NOTE: UID/GID downgrade will be delayed because of --client, --pull, or --up-delay
2020-05-08T18:07:37.244Z        INFO    openvpn: Fri May  8 18:07:37 2020 WARNING: 'link-mtu' is used inconsistently, local='link-mtu 1570', remote='link-mtu 1542'
2020-05-08T18:07:37.244Z        INFO    openvpn: Fri May  8 18:07:37 2020 WARNING: 'cipher' is used inconsistently, local='cipher AES-256-CBC', remote='cipher BF-CBC'
2020-05-08T18:07:37.244Z        INFO    openvpn: Fri May  8 18:07:37 2020 WARNING: 'auth' is used inconsistently, local='auth SHA256', remote='auth SHA1'
2020-05-08T18:07:37.244Z        INFO    openvpn: Fri May  8 18:07:37 2020 WARNING: 'keysize' is used inconsistently, local='keysize 256', remote='keysize 128'
2020-05-08T18:07:37.244Z        INFO    openvpn: Fri May  8 18:07:37 2020 [7104ea6f80145f452864558773f99167] Peer Connection Initiated with [AF_INET]104.200.151.74:1197
2020-05-08T18:07:43.513Z        INFO    openvpn: Fri May  8 18:07:43 2020 TUN/TAP device tun0 opened
2020-05-08T18:07:43.514Z        INFO    openvpn: Fri May  8 18:07:43 2020 /sbin/ip link set dev tun0 up mtu 1500
2020-05-08T18:07:43.516Z        INFO    openvpn: Fri May  8 18:07:43 2020 /sbin/ip addr add dev tun0 local 10.39.10.6 peer 10.39.10.5
2020-05-08T18:07:43.524Z        INFO    openvpn: Fri May  8 18:07:43 2020 UID set to nonrootuser
2020-05-08T18:07:43.524Z        INFO    openvpn: Fri May  8 18:07:43 2020 Initialization Sequence Completed
2020-05-08T18:08:08.760Z        ERROR   openvpn: signal: killed
2020-05-08T18:08:08.760Z        INFO    openvpn: starting
2020-05-08T18:08:08.760Z        INFO    openvpn configurator: starting openvpn
2020-05-08T18:08:08.761Z        ERROR   close |0: file already closed`

Let me know if I should open a separate issue for this

**Comment by qdm12** [comment]
Yes please open another issue and attach the full log. Also ensure you're running the latest Docker image, as I may had broken it recently. I'll probably do another release too soon in case I break `:latest`.
